### PR TITLE
feat: adds unit testing

### PR DESCRIPTION
**Pull request by stutirao**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/atoms/badge/__tests__/Badge.test.tsx [hunk 1]
```diff
@@ -29,3 +31,29 @@ describe('Badge component', () => {
 
   testHelper(mapper, testFunc);
 });
+
```
Comment (riyalohia) [inline core/components/atoms/badge/__tests__/Badge.test.tsx:34]: Add a test for `span` element.

File: core/components/atoms/pills/__tests__/Pills.test.tsx [hunk 2]
```diff
@@ -29,3 +30,32 @@ describe('Pills component', () => {
 
   testHelper(mapper, testFunc);
 });
+
+describe('Pills component', () => {
+  it('renders children', () => {
+    const { getByTestId } = render(<Pills subtle={true} appearance="secondary">{'Design System'}</Pills>);
+    expect(getByTestId('DesignSystem-Pills').textContent).toMatch('Design System');
+    expect(getByTestId('DesignSystem-Pills')).toHaveClass('Pills');
```
Comment (riyalohia) [inline core/components/atoms/pills/__tests__/Pills.test.tsx]: Remove this test and add a test for `span` element in a different describe block.

File: core/components/atoms/progressBar/__tests__/ProgressBar.test.tsx [hunk 3]
```diff
@@ -23,3 +24,16 @@ describe('ProgressBar component', () => {
...
   testHelper(mapper, testFunc);
 });
+
+describe('ProgressBar component', () => {
+  it('renders style', () => {
+    const { getByTestId } = render(<ProgressBar value={50} max={100} />);
+    expect(getByTestId('DesignSystem-ProgressBar')).toHaveStyle({ width: '50%' });
+
+  });
+  it('renders classes', () => {
```
Comment (riyalohia) [inline core/components/atoms/progressBar/__tests__/ProgressBar.test.tsx]: Remove this test.

File: core/components/atoms/progressBar/__tests__/ProgressBar.test.tsx [hunk 4]
```diff
@@ -23,3 +24,16 @@ describe('ProgressBar component', () => {
 
   testHelper(mapper, testFunc);
 });
+
+describe('ProgressBar component', () => {
+  it('renders style', () => {
+    const { getByTestId } = render(<ProgressBar value={50} max={100} />);
+    expect(getByTestId('DesignSystem-ProgressBar')).toHaveStyle({ width: '50%' });
```
Comment (riyalohia) [inline core/components/atoms/progressBar/__tests__/ProgressBar.test.tsx]: Add test for value < 0 and value > max

File: core/components/atoms/progressRing/__tests__/ProgressRing.test.tsx [hunk 5]
```diff
@@ -26,3 +27,38 @@ describe('ProgressRing component', () => {
...
   testHelper(mapper, testFunc);
 });
+
+describe('ProgressRing component', () => {
+  // const circumference = 40 * Math.PI;
+  // const strokeDashoffsetNumber = circumference - 50 / 100 * circumference;
+  it('should contain an SVG with the svg class, and a circle with the circle class', () => {
+    const { getByTestId } = render(<ProgressRing value={50} max={100} size="regular" />);
+    expect(getByTestId('DesignSystem-ProgressRing').tagName).toEqual('svg');
+    expect(getByTestId('DesignSystem-ProgressRing')).toHaveAttribute('viewBox', '0 0 50 50');
```
Comment (riyalohia) [inline core/components/atoms/progressRing/__tests__/ProgressRing.test.tsx]: Remove viewBox test.

File: core/components/atoms/progressRing/__tests__/ProgressRing.test.tsx [hunk 6]
```diff
@@ -26,3 +27,38 @@ describe('ProgressRing component', () => {
...
+    expect(getByTestId('DesignSystem-ProgressRing--Circle').tagName).toEqual('circle');
+    expect(getByTestId('DesignSystem-ProgressRing--Circle')).toHaveClass('Ring-indicator');
```
Comment (riyalohia) [inline core/components/atoms/progressRing/__tests__/ProgressRing.test.tsx]: Remove `Ring-indicator` class test

File: core/components/atoms/progressRing/__tests__/ProgressRing.test.tsx [hunk 7]
```diff
@@ -26,3 +27,38 @@ describe('ProgressRing component', () => {
...
+    expect(getByTestId('DesignSystem-ProgressRing--Circle')).toHaveAttribute('cx', '25');
+    expect(getByTestId('DesignSystem-ProgressRing--Circle')).toHaveAttribute('cy', '25');
+    expect(getByTestId('DesignSystem-ProgressRing--Circle')).toHaveAttribute('r', '20');
+    expect(getByTestId('DesignSystem-ProgressRing--Circle')).toHaveAttribute('fill', 'none');
```
Comment (riyalohia) [inline core/components/atoms/progressRing/__tests__/ProgressRing.test.tsx]: Remove `fill` test.

File: core/components/atoms/statusHint/__tests__/StatusHint.test.tsx [hunk 8]
```diff
@@ -26,3 +27,27 @@ describe('StatusHint component', () => {
 
   testHelper(mapper, testFunc);
 });
+
+describe('StatusHint component', () => {
+  it('renders children', () => {
+    const { getByTestId } = render(<StatusHint appearance="info">{'Design System'}</StatusHint>);
+    expect(getByTestId('DesignSystem-StatusHint')).toHaveClass('StatusHint');
```
Comment (riyalohia) [inline core/components/atoms/statusHint/__tests__/StatusHint.test.tsx]: Remove this test.

File: core/components/atoms/statusHint/__tests__/StatusHint.test.tsx [hunk 9]
```diff
@@ -26,3 +27,27 @@ describe('StatusHint component', () => {
...
+  });
+  it('renders children', () => {
+    const { getByTestId } = render(<StatusHint appearance="info">{'Design System'}</StatusHint>);
+    expect(getByTestId('DesignSystem-StatusHint--Text').textContent).toMatch('Design System');
+  });
+  it('renders children', () => {
+    const { getByTestId } = render(<StatusHint appearance="info">{'Design System'}</StatusHint>);
+    expect(getByTestId('DesignSystem-StatusHint--Icon')).toHaveClass('StatusHint-icon');
```
Comment (riyalohia) [inline core/components/atoms/statusHint/__tests__/StatusHint.test.tsx]: Remove this test and add test for `span` element.

**Review by riyalohia — CHANGES_REQUESTED** [review]

File: core/components/atoms/paragraph/__tests__/Paragraph.test.tsx [hunk 10]
```diff
@@ -27,3 +28,19 @@ describe('Paragraph component', () => {
 
   testHelper(mapper, testFunc);
 });
+describe('Paragraph component', () => {
+  it('renders children', () => {
```
Comment (riyalohia) [inline core/components/atoms/paragraph/__tests__/Paragraph.test.tsx:33]: Add a test for `p` tag.

File: core/components/atoms/subheading/__tests__/Subheading.test.tsx [hunk 11]
```diff
@@ -27,3 +28,18 @@ describe('Subheading component', () => {
 
   testHelper(mapper, testFunc);
 });
+describe('Subheading component', () => {
+  it('renders children', () => {
+    const { getByTestId } = render(<Subheading>{'Subheading'}</Subheading>);
+    expect(getByTestId('DesignSystem-Subheading')).toHaveTextContent('Subheading');
```
Comment (riyalohia) [inline core/components/atoms/subheading/__tests__/Subheading.test.tsx:36]: Add test for `h4` tag.

**Review by riyalohia — CHANGES_REQUESTED** [review]

**Comment by riyalohia** [comment]
@stutirao Add `class override` test for each component.

**Review by riyalohia — APPROVED** [review]

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**